romancal/regtest/test_mos_pipeline.py

```diff
@@ -128,3 +129,52 @@ def test_level3_mos_pipeline(rtdata, ignore_asdf_paths):
         "DMS86 MSG: Testing completion of resample in the Level 3 image output......."
         + passfail(model.meta.cal_step.resample == "COMPLETE")
     )
+
+
+@pytest.mark.bigdata
+@pytest.mark.soctests
+@metrics_logger("DMS373")
+def test_hlp_mosaic_pipeline(rtdata, ignore_asdf_paths):
+    """Tests for level 3 mosaic requirements DMS373"""
+
+    cal_files = [
+        "WFI/image/r0000101001001001001_01101_0001_WFI01_cal.asdf",
+        "WFI/image/r0000101001001001001_01101_0002_WFI01_cal.asdf",
+        "WFI/image/r0000101001001001001_01101_0003_WFI01_cal.asdf",
+    ]
+
+    asn = asn_from_list(
+        cal_files, product_name="r0099101001001001001_r274dp63x31y81_F158_prompt_visit"
+    )
+    asn["target"] = "r274dp63x31y81"
+    asn_name = "L3_mosaic_asn.json"
+    _, serialized = asn.dump()
+    with open(asn_name, "w") as f:
+        f.write(serialized)
```

Comment:
```suggestion
```
The association is being saved to `L3_mosaic_asn.json` which is being overwritten on line 158.